worker/provisioner/provisioner_task.go

```diff
@@ -168,6 +162,13 @@ func (task *provisionerTask) Wait() error {
 
 func (task *provisionerTask) loop() error {
 
+	// Get existing machine distributions.
+	err := task.populateAvailabilityZoneMachines()
```

Comment:
I was wondering whether there was anything that expected `availabilityZoneMachines` to be populated by the time the worker is returned, but from going through the code it looks like the only things that use it are called from loop, so seems all good.